### PR TITLE
exclude ipc scheme from port check

### PR DESCRIPTION
**Pull request by JayBitron**
Previous PR https://github.com/bitcoin/bitcoin/pull/22087 cause new error, makes it impossible to use ipc protocol using zmq, this patch will exclude port checking on ipc urls.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1261,7 +1261,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         for (const std::string& socket_addr : args.GetArgs(port_option)) {
             std::string host_out;
             uint16_t port_out{0};
-            if (!SplitHostPort(socket_addr, port_out, host_out)) {
+            if (socket_addr.rfind("ipc:", 0) != 0 && !SplitHostPort(socket_addr, port_out, host_out)) {
```
Comment (luke-jr) [inline src/init.cpp:1264]: `rfind` is a very weird choice here. Suggest using [`compare`](https://cplusplus.com/reference/string/string/compare/)

**Review by luke-jr — CHANGES_REQUESTED** [review]

**Comment by luke-jr** [comment]
nit: Rebasing onto bbbf89a9de0757c44880495244f90967f7147c0d would enable a clean merge to 25.x also

File: src/init.cpp (hunk 1 above)
Comment (JayBitron) [reply src/init.cpp:1264]: `compare` throws an exception if the input string is lower than 4 character, but since this a conflict PR, please close it.

**Comment by pinheadmz** [comment]
See also https://github.com/bitcoin/bitcoin/pull/27679 which closes the same issue and has tests

**Pull request closed without merging**